app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -332,40 +360,72 @@ class ParkingViewModel(
     parkingRepository.addReport(
         report,
         onSuccess = {
-          // Check thresholds for reporting to `ReportedObjectRepository`
-          // i.e if it's max severity add it to the maximum severity count and check if it doesn't
-          // exceed the maximum severity threshold. Otherwise, check if it doesn't exceed
-          // the standard threshold
-          if ((report.reason.severity == MAX_SEVERITY &&
-              selectedParking.nbMaxSeverityReports >= NB_REPORTS_MAXSEVERITY_THRESH) ||
-              (selectedParking.nbReports >= NB_REPORTS_THRESH)) {
-            reportedObjectRepository.addReportedObject(
-                ReportedObject(
-                    objectUID = selectedParking.uid,
-                    reportUID = report.uid,
-                    reason = ReportReason.Parking(report.reason),
-                    userUID = user.public.userId,
-                    objectType = ReportedObjectType.PARKING),
-                onSuccess = {},
-                onFailure = {})
-          }
-
-          // Update the local reports and parking metrics
-          _selectedParkingReports.update { currentReports ->
-            currentReports?.plus(it) ?: listOf(it)
-          }
-          if (report.reason.severity == MAX_SEVERITY) {
-            selectedParking.nbMaxSeverityReports += 1
-          }
-          selectedParking.nbReports += 1
-          parkingRepository.updateParking(selectedParking, {}, {})
-          Log.d("ParkingViewModel", "Report added successfully")
+          val newReportedObject =
+              ReportedObject(
+                  objectUID = selectedParking.uid,
+                  reportUID = report.uid,
+                  nbOfTimesReported = selectedParking.nbReports + 1,
+                  nbOfTimesMaxSeverityReported =
+                      if (report.reason.severity == 3) selectedParking.nbMaxSeverityReports + 1
+                      else selectedParking.nbMaxSeverityReports,
+                  userUID = user.public.userId,
+                  objectType = ReportedObjectType.PARKING,
+              )
+
+          reportedObjectRepository.getDocumentReferenceByObjectUID(
+              objectUID = selectedParking.uid,
+              onSuccess = { documentRef ->
```

Comment:
Note that we're in ParkingViewModel here. This deals with Reports (which are a subcollection of a Parking) and not an object, which is a separate thing entirely. You're not the first to get confused by this, may change the names in a future PR.
I didn't realize that making the changes inside the ViewModel instead of the Firestore made so much of a difference here, though it makes sense in retrospect. Shouldn't be too difficult a fix. Apologies for this and thanks for bringing to my attention